### PR TITLE
Add override_second and set_override_second APIs

### PR DESCRIPTION
**Pull request by InfRandomness**
Hey everyone 👋 

I'm opening this PR because I was suggested by a Discord member to create this API as a newcommer PR to make.

I've created the override_second and set_override_second APIs, both equally using their "second" counterpart APIs

I only created the APIs I thought were useful, being unsure if the other APIs such as 
https://github.com/SanderMertens/flecs/blob/master/include/flecs/addons/cpp/mixins/entity/builder.hpp#L375-L384
or
https://github.com/SanderMertens/flecs/blob/master/include/flecs/addons/cpp/mixins/entity/builder.hpp#L386-L395
also need replication.

:)

### CODE REVIEW

File: include/flecs/addons/cpp/mixins/entity/builder.hpp [hunk 1]
```diff
@@ -394,6 +394,35 @@ struct entity_builder : entity_view {
...
 
+    /** Mark id for auto-overriding.
+     * When an entity inherits from a base entity (using the IsA relationship)
+     * any ids marked for auto-overriding on the base will be overridden
+     * automatically by the entity.
+     *
+     * @param id The id to mark for overriding.
+     */
+    Self& override_second(flecs::id_t id) {
+        return this->add_second(ECS_OVERRIDE | id);
```
Comment (SanderMertens) [inline include/flecs/addons/cpp/mixins/entity/builder.hpp:405]: I don't think this works, the `add_second` method requires a template argument:
```cpp
    template<typename Second>
    Self& add_second(flecs::entity_t first) {
        return this->add(first, _::cpp_type<Second>::id(this->m_world));
    }
```

File: include/flecs/addons/cpp/mixins/entity/builder.hpp [hunk 2]
```diff
@@ -394,6 +394,35 @@ struct entity_builder : entity_view {
...
+    template <typename Second>
+    Self& override_second(flecs::entity_t first) {
+        return this->override(first, _::cpp_type<Second>::id(this->m_world));
+    }
+
+    /** Mark pair for auto-overriding.
+     * @param first The first element of the pair.
+     * @param second The second element of the pair.
+     */
+    Self& override_second(flecs::entity_t first, flecs::entity_t second) {
```
Comment (SanderMertens) [inline include/flecs/addons/cpp/mixins/entity/builder.hpp:422]: I don't think this method is necessary. The `*_second` functions are necessary when the first element of a pair is an id, and the second element of a pair is a type. In this case both elements are ids, which makes this function a duplicate of:

```cpp
    Self& override(flecs::entity_t first, flecs::entity_t second) {
        return this->override(ecs_pair(first, second));
    }
```

File: include/flecs/addons/cpp/mixins/entity/builder.hpp [hunk 3]
```diff
@@ -501,6 +530,17 @@ struct entity_builder : entity_view {
         return to_base();  
     }
 
+     /** Set component, mark component for auto-overriding.
+      * @see override(flexs::id_t id)
+      *
+      * @tparam T the component to set and for which to add the OVERRIDE flag
+      */
+     template<typename T>
+     Self& set_override_second(const T& val) {
```
Comment (SanderMertens) [inline include/flecs/addons/cpp/mixins/entity/builder.hpp:539]: Methods with `_second` operate on pairs, which this method doesn't. You're missing a `flecs::entity_t first` argument. Additionally, to make sure `set` works in all cases (by value, by ref, move ref etc) a number of overloads have to be implemented, see `set_override` for examples.

**Review by SanderMertens — CHANGES_REQUESTED** [review]

**Comment by InfRandomness** [comment]
I will correct these today (hopefully), 👍

File: include/flecs/addons/cpp/mixins/entity/builder.hpp (hunk 1 above)
Comment (InfRandomness) [reply include/flecs/addons/cpp/mixins/entity/builder.hpp:405]: I'm not sure what I should be passing over to add_second 🤔

File: include/flecs/addons/cpp/mixins/entity/builder.hpp (hunk 3 above)
Comment (InfRandomness) [reply include/flecs/addons/cpp/mixins/entity/builder.hpp:539]: are all of those overloads needed?

File: include/flecs/addons/cpp/mixins/entity/builder.hpp (hunk 1 above)
Comment (MewSoul) [reply include/flecs/addons/cpp/mixins/entity/builder.hpp:405]: Probably something like that:
```cpp
    template <typename Second>
    Self& override_second(flecs::id_t id) {
        return this->add_second<Second>(ECS_OVERRIDE | id);
    }
```

**Comment by SanderMertens** [comment]
Closing PR for inactivity.

**Pull request closed without merging**